clients/adapter/src/main/java/org/eclipse/hono/client/upstream/CommandResponseSender.java

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.client.upstream;
+
+import io.opentracing.SpanContext;
+import io.vertx.core.Future;
+import io.vertx.proton.ProtonDelivery;
+
+/**
+ * A client for publishing a device's response to a command received from a downstream application.
+ */
+public interface CommandResponseSender {
```

Comment:
The interfaces in the `upstream` package still depend heavily on AMQP. Do we change them completely again if we want to support Kafka for commands? I think that if you use Kafka for downstream messaging, you might want to use Hono in the future entirely without AMQP.

Reply:
Yes, that is true. We might want to create `Command` and `CommandResponse` to be independent from the underlying transport.